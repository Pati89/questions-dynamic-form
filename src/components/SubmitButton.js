import PropTypes from "prop-types";
import styles from "../styles/components/SubmitButton.module.css";

const SubmitButton = ({ disabled, text, onClick, type }) => {
  return (
    <button
      onClick={onClick}
      disabled={disabled}
      type={type}
      className={styles.submitButton}
      data-testid="submit-button"
    >
      {text}
    </button>
  );
};

SubmitButton.defaultProps = {
  type: "submit",
  text: "Submit",
};

SubmitButton.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  type: PropTypes.oneOf(["submit", "button", "reset"]),
};

export default SubmitButton;
