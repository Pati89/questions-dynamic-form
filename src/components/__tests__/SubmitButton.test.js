import SubmitButton from "../SubmitButton";
import { render, screen, fireEvent } from "@testing-library/react";

const props = {
  onClick: () => {},
};

describe("SubmitButton", () => {
  test("should render a SubmitButton", () => {
    render(<SubmitButton {...props} />);
    const button = screen.getByTestId("submit-button");
    expect(button).toBeTruthy();
  });

  test("should render default text", () => {
    render(<SubmitButton {...props} />);
    const buttonText = screen.getByText("Submit");
    expect(buttonText).toBeTruthy();
  });

  test("should render text", () => {
    render(<SubmitButton {...props} text="test" />);
    const buttonText = screen.getByText("test");
    expect(buttonText).toBeTruthy();
  });

  test("should disable the button", () => {
    render(<SubmitButton {...props} disabled />);
    const button = screen.getByTestId("submit-button");
    expect(button).toBeDisabled();
  });

  test("should trigger onClick function", () => {
    const spy = jest.fn();
    render(<SubmitButton onClick={spy} />);
    const button = screen.getByTestId("submit-button");
    fireEvent.click(button);
    expect(spy).toHaveBeenCalled();
  });

  test("should render a SubmitButton match snapshot", () => {
    render(<SubmitButton />);
    const button = screen.getByTestId("submit-button");
    expect(button).toMatchSnapshot();
  });
});
