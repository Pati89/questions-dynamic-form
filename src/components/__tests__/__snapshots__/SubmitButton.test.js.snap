// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton should render a SubmitButton match snapshot 1`] = `
<button
  class="submitButton"
  data-testid="submit-button"
  type="submit"
>
  Submit
</button>
`;
